stub_uploader/get_version.py

```diff
@@ -62,13 +66,26 @@ def compute_incremented_version(
     # version will satisfy the version_spec (defined precisely by the `compatible`
     # specifier below). This allows users to have expectations of what a stub package
     # will contain based on the upstream version they're targeting.
-    if version_spec.endswith(".*"):
+    m = POST_SPEC_RE.match(version_spec)
+    if m:
```

Comment:
We could use `Version(version_spec.removesuffix(".*")).post is not None` instead of regex parsing. E.g. I think this doesn't handle https://peps.python.org/pep-0440/#implicit-post-releases